Export primary HGNC transcript in variantS export file

northwestwitch opened the pull request:
This PR adds a functionality:
- merges 3 columns relative to the canonical transcript into one: `Canonical_transcript|HGVS|protein_change`
- creates another column for the primary transcript:  `Primary_transcript|HGVS|protein_change`
- Fixes a bug caused by using default genome build 37 when exporting these info from variants page

Partial fix #2398 and possible fix #2429 

**How to test**:
1. Go to variantS page, filter and export variants.
2. Check exported transcripts against transcripts specified on each variant's page

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

bjhall commented on scout/server/blueprints/variants/controllers.py:
```diff
@@ -445,47 +444,77 @@ def variant_export_lines(store, case_obj, variants_query):
-    for item in [gene_ids, gene_names, canonical_txs, hgvs_c, pt_c]:
-        gene_info.append(";".join(str(x) for x in item))
+    for item in [gene_ids, gene_names, canonical_txs, primary_txs]:
+        gene_info.append("-".join(str(x) for x in item))
```
I think you need to join with `" - "` (with spaces). Since gene symbols can contain dashes things like `CYP1B1-CYP1B1-AS1` become ambiguous.

bjhall commented:
This looks much better now! It even works in Swedish Excel! 

However I am getting almost nothing in the canonical field. And when there is no primary transcript I get nothing at all (which is expected, but a bit sad :) ).

![bild](https://user-images.githubusercontent.com/11991860/110943481-8ace0a00-833b-11eb-84cd-6c3808842f50.png)

northwestwitch commented:
Right! I chose the dumbest separator ever, sorry! Regarding the very few canonical transcripts I am quite sure that this is due to the genome build bugfix. I'll think about it and get back to you with a fix, thanks!

northwestwitch commented:
@bjhall I've changed the inner separators to "/" and the gene separators to " | ". I've also changed how the canonical transcripts are collected and you should be able to see more of them now!

bjhall commented:
Yes now this looks much better! I still see very few (maybe 1 out of 100) variants where I get nothing in either the canonical or primary column. But these appear to be when 1) there is no defined primary/canonical transcript or 2) when the canonical transcript is some weird non-coding thing. 

Maybe leaving those corner cases unfixed is fine, since the export function is "for research only", right?

northwestwitch commented:
> Maybe leaving those corner cases unfixed is fine, since the export function is "for research only", right?

Yes! I see the same with build 37 and I think it's fine! If you think it's fine could you approve the PR? So we'll have it in the next release!

northwestwitch commented:
Thanks @bjhall, merging!
